### DISCUSSION TITLE
Mirror transform for ospInstance

### DISCUSSION THREAD

**Discussion by anskwagh** [RenderKit/ospray#580] [Q&A] [answered]
Hello,

I am able to create ospInstance using groups.
I have created instance1 using ospGroups and applied some transform to it.
I want to create instance2 by copying/using instance1 (not using ospGroup) and add extra transform to it (along with previous transform of ospInstance1). Do you know if thats possible and how can it be done?

**Comment by johguenther** [comment, marked as answer]
If I understand your question correctly then you are asking for multi-level instancing, which is currently not supported (but a common feature request, see e.g. https://github.com/ospray/ospray/issues/547). Until we add this, the workaround is to pre-process the transformations in the application (or the scene graph) by combining the transformation of instance1 and instance2. This is how we do that in [OSPRay Studio](https://www.ospray.org/ospray_studio/), see https://github.com/ospray/ospray_studio/blob/master/sg/visitors/RenderScene.h#L123-L159.
